### PR TITLE
fix(docker): require explicit REDIS_PASSWORD + REDIS_TOKEN, drop wm-local-token default (#3804)

### LINKED ISSUE #3804: Docker Redis runs with no password and redis-rest defaults to hardcoded token 'wm-local-token'
## Summary

The Docker Compose configuration runs Redis with no password and the `redis-rest` (Upstash SRH) proxy with a hardcoded default token `wm-local-token`. Any container on the `worldmonitor` Docker network can connect to Redis without credentials. If the `redis-rest` port is ever exposed beyond `127.0.0.1`, the token needed to use it is already publicly known.

## Evidence

**`docker-compose.yml`:**
```yaml
redis:
  image: redis:7-alpine
  command: redis-server --maxmemory 256mb --maxmemory-policy allkeys-lru
  # No requirepass — Redis accepts all connections without authentication

redis-rest:
  environment:
    SRH_TOKEN: "${REDIS_TOKEN:-wm-local-token}"   # default is a known public string
    SRH_CONNECTION_STRING: "redis://redis:6379"
  ports:
    - "127.0.0.1:8079:80"                          # currently localhost-only

worldmonitor:
  environment:
    UPSTASH_REDIS_REST_TOKEN: "${REDIS_TOKEN:-wm-local-token}"  # same default
```

## Why this matters

1. **No Redis password**: any process that gains a foothold in any Docker container on the `worldmonitor` network (e.g. via a compromised dependency or SSRF vulnerability) can read and write all Redis keys without any credential — including session tokens, rate-limit counters, and cached AIS/OpenSky data.

2. **Known default token**: If `REDIS_TOKEN` is not set (e.g. when following the getting-started guide without reading the env docs), the REST token is the publicly documented string `wm-local-token`. Anyone aware of the defaults can authenticate to the REST API if the port is ever exposed.

3. **Pattern risk**: the `127.0.0.1:8079:80` binding prevents external access today, but any future `docker-compose.override.yml` or port-forwarding change would immediately expose an unauthenticated REST interface.

## Attack or failure scenario

1. Attacker exploits any SSRF or container escape in the `worldmonitor` app container.
2. Issues Redis commands directly to `redis:6379` — no password required.
3. Can read all cached session tokens, forge rate-limit counters to zero, poison AIS snapshot caches with fabricated vessel positions.

## Root cause

Redis is run in "trusted network" mode appropriate only for isolated local dev, but the compose file is also used for self-hosted production deployments.

## Recommended fix

1. Add a Redis password to the compose file:
   ```yaml
   redis:
     command: redis-server --requirepass "${REDIS_PASSWORD:?REDIS_PASSWORD required}" --maxmemory 256mb --maxmemory-policy allkeys-lru
   ```
2. Update `SRH_CONNECTION_STRING` to include the password:
   ```yaml
   SRH_CONNECTION_STRING: "redis://:${REDIS_PASSWORD}@redis:6379"
   ```
3. Remove the `wm-local-token` default from `SRH_TOKEN` and `UPSTASH_REDIS_REST_TOKEN` — require explicit env var.
4. Add `REDIS_PASSWORD`, `REDIS_TOKEN` to the `.env.example` with placeholder values and installation docs.

## Acceptance criteria

- [ ] Redis requires a password for all connections in the Docker Compose setup.
- [ ] `SRH_TOKEN` has no hardcoded default; the relay fails to start if `REDIS_TOKEN` is unset.
- [ ] `.env.example` documents `REDIS_PASSWORD` and `REDIS_TOKEN` as required fields.

## Suggested labels
security, docker

## Priority
P2

## Severity
Medium — unauthenticated Redis in a Docker network used by a public-facing service.

## Confidence
Confirmed — `docker-compose.yml` verified.

### PR DESCRIPTION
**Pull request by koala73**
## Summary

Closes #3804.

Self-hosted Docker stack shipped two interlocking security defaults that made the deployment one hostile `docker-compose.override.yml` away from "publicly authenticated by a known string":

1. **Redis ran without `--requirepass`** — any container added to the `worldmonitor` docker network could read/write all cache entries (session tokens, rate-limit counters, AIS snapshots) without credentials. The `redis-rest` REST proxy held the only credential gate.
2. **`SRH_TOKEN` and `UPSTASH_REDIS_REST_TOKEN` both defaulted to `wm-local-token`** — `${REDIS_TOKEN:-wm-local-token}`. `SELF_HOSTING.md:120` and `scripts/run-seeders.sh:9` *reinforced* the default with copy-pasteable snippets. A fresh clone that flipped the `redis-rest` binding from `127.0.0.1:8079:80` to `0.0.0.0:8079:80` (one-line override) instantly exposed an authenticated REST interface authenticated by a token anyone reading the docs already knew.

Today's `127.0.0.1` binding contains external exposure, but the threat model is brittle — and the docs were actively encouraging the brittle path.

## Changes

- **`docker-compose.yml`** — `${VAR:?...}` fail-closed expansion for `REDIS_PASSWORD` and `REDIS_TOKEN` on every service that touches them. Affected containers refuse to start with a helpful error pointing at `openssl rand -hex 32`. Redis gains `--requirepass "${REDIS_PASSWORD:?...}"`; `redis-rest` plumbs the password into `SRH_CONNECTION_STRING` (`redis://:${REDIS_PASSWORD}@redis:6379`) so it still reaches Redis.
- **`.env.example`** — new "Self-hosted Redis (Docker Compose only)" section documenting both vars + how to generate, distinguishing them from the Vercel-Upstash hosted block above.
- **`SELF_HOSTING.md`** — quick-start now generates all three secrets in step 2; required-env table covers all three with a migration note for existing clones; manual-seeder snippet uses `${REDIS_TOKEN:?...}` instead of the literal.
- **`scripts/run-seeders.sh`** — auto-sources `REDIS_TOKEN` from `.env`, fail-loud if neither `REDIS_TOKEN` nor `UPSTASH_REDIS_REST_TOKEN` is set instead of silently using `wm-local-token`.

## Regression test

`tests/docker-compose-no-default-secrets.test.mts` — literal-absence + structural-presence source-greps over all four files, following the static-grep recipe from PR #3778 (CORS fail-closed). The `SHIPPED_DEFAULT_PATTERNS` regex set is paired with a meta-test exercising 7 representative inputs (4 historical bad shapes + 3 acceptable shapes including prose mentions of the literal in a deprecation note) so a future "simplification" of the pattern can't silently drop coverage.

**Reverse-check performed:** temporarily reintroducing the bad default in `docker-compose.yml` fails the matching test with the `#3804` reference; restoring the fix turns it green again.

## Test plan

- [x] `npx tsx --test tests/docker-compose-no-default-secrets.test.mts` → 5/5 pass locally
- [x] Reverse-test confirms each assertion fires when its bad default is reintroduced
- [x] Pre-push hook (full test suite + version:check) passed
- [ ] Existing self-hosters need a one-time `.env` update — release notes should call this out

## Out of scope

- Doesn't touch the Vercel Upstash credential path (`UPSTASH_REDIS_REST_URL` / `UPSTASH_REDIS_REST_TOKEN` sourced from Upstash itself, unaffected by this change)
- Docker secrets (`secrets:` block) remains the documented escalation path for production deployments that don't want creds in `docker inspect`

### CODE REVIEW

File: scripts/run-seeders.sh [hunk 1]
```diff
@@ -5,13 +5,32 @@
...
+UPSTASH_REDIS_REST_URL="${UPSTASH_REDIS_REST_URL:-http://localhost:8079}"
+if [ -z "${UPSTASH_REDIS_REST_TOKEN:-}" ] && [ -n "${REDIS_TOKEN:-}" ]; then
+  UPSTASH_REDIS_REST_TOKEN="$REDIS_TOKEN"
+fi
+if [ -z "${UPSTASH_REDIS_REST_TOKEN:-}" ]; then
+  echo "ERROR: REDIS_TOKEN (or UPSTASH_REDIS_REST_TOKEN) is required." >&2
+  echo "       Generate with: openssl rand -hex 32, then add to .env" >&2
+  echo "       See SELF_HOSTING.md → Required Environment Variables." >&2
+  exit 1
+fi
```
[the remark this answers is not in the document]
Comment (koala73) [reply scripts/run-seeders.sh:37]: Good catch — confirmed the precedence inversion in commit ba1e55882. The Docker-only header on this script makes `REDIS_TOKEN` the unambiguous winner; `UPSTASH_REDIS_REST_TOKEN` is now only a fallback if `REDIS_TOKEN` is missing. Also added two source-grep assertions in `tests/docker-compose-no-default-secrets.test.mts` (one asserting the bad gate is gone, one asserting the bare `if [ -n REDIS_TOKEN ]` form is present) — reverse-checked that reintroducing the old AND-conditional trips the test with this PR's number in the failure message.

File: docker-compose.yml [hunk 2]
```diff
@@ -86,8 +93,8 @@ services:
     ports:
       - "127.0.0.1:8079:80"
     environment:
-      SRH_TOKEN: "${REDIS_TOKEN:-wm-local-token}"
-      SRH_CONNECTION_STRING: "redis://redis:6379"
+      SRH_TOKEN: "${REDIS_TOKEN:?REDIS_TOKEN required — generate with: openssl rand -hex 32}"
+      SRH_CONNECTION_STRING: "redis://:${REDIS_PASSWORD}@redis:6379"
```
[the remark this answers is not in the document]
Comment (koala73) [reply docker-compose.yml]: Fixed in ba1e55882. Switched the `SRH_CONNECTION_STRING` expansion to the same fail-closed `${REDIS_PASSWORD:?…}` form, with a comment explicitly calling out it's defense-in-depth against a future PR removing `--requirepass` or copying the string elsewhere. Also tightened the regression test to use a negative-lookahead regex (`$\{REDIS_PASSWORD(?![:?])` / `$\{REDIS_TOKEN(?![:?])`) so ANY bare expansion anywhere in the file — not just the originally-flagged line — trips the assertion. Reverse-check confirmed the new test fires when the bare form is reintroduced.
